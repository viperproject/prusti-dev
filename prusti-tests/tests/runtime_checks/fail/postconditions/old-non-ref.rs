//@run: 101
//@compile-flags: -Pcheck_overflows=false
use prusti_contracts::*;

#[derive(Clone)]
struct Something {
    pub field: i32,
}

// has to fail, since x.field in specification refers
// to old state too
#[trusted]
#[insert_runtime_check]
#[ensures(old(x.field) + 4 == x.field)]
fn nonsense(mut x: Something) {
    x.field = x.field + 4;
}

fn main() {
    let s = Something {
        field: 1,
    };
    nonsense(s);
}
