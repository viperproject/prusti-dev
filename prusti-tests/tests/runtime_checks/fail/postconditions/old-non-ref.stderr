thread 'main' panicked at $DIR/old-non-ref.rs:15:1:
Prusti Runtime Checks: Contract #[ensures(old(x.field) + 4 == x.field)] was violated at runtime
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
